### PR TITLE
Implement ALIAS NODE keyword locations

### PR DESCRIPTION
**Pull request by ydah**

### CODE REVIEW

File: node_dump.c [hunk 1]
```diff
@@ -924,6 +924,7 @@ dump_node(VALUE buf, VALUE indent, int comment, const NODE * node)
         F_NODE(nd_1st, RNODE_ALIAS, "new name");
         LAST_NODE;
```
Comment (yui-knk) [inline node_dump.c:925]: Could you move `LAST_NODE;` one line below.
Output of `--dump=p` is bit broken with current implementation (no `|` between `nd_2nd` and `keyword_loc`):

```
$ ./miniruby --dump=p -e 'alias :a :b'
###########################################################
## Do NOT use this node dump for any purpose other than  ##
## debug and research.  Compatibility is not guaranteed. ##
###########################################################

# @ NODE_SCOPE (id: 3, line: 1, location: (1,0)-(1,11))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_ALIAS (id: 2, line: 1, location: (1,0)-(1,11))*
#     +- nd_1st:
#     |   @ NODE_SYM (id: 0, line: 1, location: (1,6)-(1,8))
#     |   +- string: :a
#     +- nd_2nd:
#         @ NODE_SYM (id: 1, line: 1, location: (1,9)-(1,11))
#         +- string: :b
#     +- keyword_loc: (1,0)-(1,5)
```

Comment (ydah) [reply node_dump.c]: Thank you catching this! I fixed it.

```
# @ NODE_SCOPE (id: 3, line: 1, location: (1,0)-(1,11))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_ALIAS (id: 2, line: 1, location: (1,0)-(1,11))*
#     +- nd_1st:
#     |   @ NODE_SYM (id: 0, line: 1, location: (1,6)-(1,8))
#     |   +- string: :a
#     +- nd_2nd:
#     |   @ NODE_SYM (id: 1, line: 1, location: (1,9)-(1,11))
#     |   +- string: :b
#     +- keyword_loc: (1,0)-(1,5)
```

**Comment by yui-knk** [comment]
@ydah After merge https://github.com/ruby/ruby/pull/11523, this PR conflicts with master branch. Could you rebase this PR?

**Comment by ydah** [comment]
> @ydah After merge #11523, this PR conflicts with master branch. Could you rebase this PR?

I did it🙏

**Pull request merged**